packages/salesforcedx-vscode-core/test/jest/commands/retrieveExecutor.test.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { RetrieveExecutor } from '../../../src/commands/RetrieveExecutor';
+import { WorkspaceContext } from '../../../src/context/workspaceContext';
+import { SourceTrackingService } from '../../../src/services';
+
+const dummyProjectPath = '/a/project/path';
+jest.mock('@salesforce/source-deploy-retrieve', () => {
+  return {
+    ...jest.requireActual('@salesforce/source-deploy-retrieve'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ComponentSet: jest.fn().mockImplementation(() => {
+      return {
+        retrieve: jest.fn().mockImplementation(() => {
+          return { pollStatus: jest.fn() };
+        })
+      };
+    })
+  };
+});
+
+jest.mock('@salesforce/salesforcedx-utils-vscode', () => {
+  return {
+    ...jest.requireActual('@salesforce/salesforcedx-utils-vscode'),
+    getRootWorkspacePath: () => dummyProjectPath,
+    ChannelService: jest.fn().mockImplementation(() => {
+      return {};
+    }),
+    TelemetryService: { getInstance: jest.fn() },
+    TelemetryBuilder: jest.fn()
+  };
+});
+
+jest.mock('../../../src/messages', () => {
+  return { loadMessageBundle: jest.fn(), nls: { localize: jest.fn() } };
+});
+
+jest.mock('../../../src/commands/util/postconditionCheckers');
+
+jest.mock('../../../src/sfdxProject/sfdxProjectConfig');
+
+describe('Retrieve Executor', () => {
+  const dummyProcessCwd = '/';
+  const workspaceContextGetInstanceSpy = jest.spyOn(
+    WorkspaceContext,
+    'getInstance'
+  );
+  const mockWorkspaceContext = { getConnection: jest.fn() } as any;
+  const createSourceTrackingSpy = jest.spyOn(
+    SourceTrackingService,
+    'createSourceTracking'
+  );
+  const dummyComponentSet = new ComponentSet();
+  const retrieveSpy = jest.spyOn(dummyComponentSet, 'retrieve');
```

Comment:
same story for the mocks here. constants are fine wrapped in the describe function